lib/modules/manager/bazel/parser.ts

```diff
@@ -80,17 +94,93 @@ const kwParams = q
         }
         return ctx;
       },
-      search: q.str<Ctx>((ctx, { value, offset }) => {
-        const parentRecord = currentFragment(ctx);
-        if (parentRecord.type === 'record' && ctx.recordKey) {
-          const key = ctx.recordKey;
-          const array = parentRecord.children[key];
-          if (array.type === 'array') {
-            array.children.push({ type: 'string', value, offset });
+      search: q.alt(
+        q.str<Ctx>((ctx, { value, offset }) => {
+          const parentRecord = currentFragment(ctx);
+          if (parentRecord.type === 'record' && ctx.recordKey) {
+            const key = ctx.recordKey;
+            const array = parentRecord.children[key];
+            if (array.type === 'array') {
+              array.children.push({ type: 'string', value, offset });
+            }
           }
-        }
-        return ctx;
-      }),
+          return ctx;
+        }),
+        q
+          .sym<Ctx>()
+          .handler(recordStartHandler)
+          .handler((ctx, { value, offset }) => {
+            const ruleFragment = currentFragment(ctx);
+            if (ruleFragment.type === 'record') {
+              ruleFragment.children['function'] = {
```

Comment:
do we need to use a reserved keyword `function` here? can we shorten to `func` so we don't need the verbose indexer Notation?